pkg/backup/compaction_job.go

```diff
@@ -610,24 +608,28 @@ func createCompactionManifest(
 	if len(tenantSpans) != 0 || len(tenantInfos) != 0 {
 		return nil, errors.New("backup compactions does not yet support range keys")
 	}
-	m, err := createBackupManifest(
-		ctx,
-		execCtx.ExecCfg(),
-		tenantSpans,
-		tenantInfos,
-		details,
-		compactChain.backupChain,
-		compactChain.allIters,
-	)
-	if err != nil {
-		return nil, err
-	}
-	m.IsCompacted = true
-	m.IntroducedSpans, err = compactIntroducedSpans(ctx, m, compactChain)
+	lastBackup := compactChain.lastBackup()
+	introducedSpans, err := compactChain.compactIntroducedSpans(ctx)
 	if err != nil {
 		return nil, err
 	}
-	return &m, nil
```

Comment:
nit: i think this logic is less error prone to to the intro of new fields to the manifest:
```
cManifest := lastBackup
cManifest.StartTime = details.StartTime
// continue setting the fields that are not inherited by the last backup
```

Reply:
just remember to nil out progress and files :)

Reply:
I think it depends on the new field right? If a field is meant to be specific to a backup (aka like the files fields), then that method would be susceptible to that bug.

That being said, I think you're right anyway since I would imagine that most newly added fields would want to be carried over in the case of compaction.